test/integration/features/basic.feature

```diff
@@ -10,6 +10,61 @@ Feature: Basic
      Then Minishift should have state "Running"
       And Minishift should have a valid IP address
 
+  Scenario: User is able to do ssh into Minishift VM
+    Given Minishift has state "Running"
+     When executing "minishift ssh echo hello"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      hello
+      """
+
+  Scenario: User is able to set custom Docker specific environment variables
+    Given Minishift has state "Running"
+     When executing "minishift ssh cat /var/lib/boot2docker/profile"
+     Then stderr should be empty
+      And exitcode should equal 0
+      And stdout should contain
+      """
+      export "FOO=BAR"
+      export "BAZ=BAT"
+      """
+
+  Scenario Outline: User can set, get, view and unset values in configuration file
+    User is able to setup persistent configuration of Minishift using "config" command
+    and its subcommands.
+    Given Minishift has state "Running"
+     When executing "minishift config set <property> <value>"
+     Then JSON config file contains key "<property>" with value "<value>"
+      And stdout of command "minishift config get <property>" is equal to "<value>"
+      And stdout of command "minishift config view --format {{.ConfigKey}}:{{.ConfigValue}}" contains "<property>:<value>"
+     When executing "minishift config unset <property>"
+     Then stdout of command "minishift config get <property>" is equal to "<nil>"
+      And JSON config file does not contain key "<property>"
+      
+  Examples: Config values to work with
+    | property  | value |
+    | disk-size | 22g   |
+    | memory    | 2222  |
+    | cpus      | 3     |
+
+  Scenario: Commands console and ip returns valid ip
+    User is able to get ip of Minishift VM with command "ip" and url adress of OpenShift instance
+    by running command "console" with flag "--url". Both commands should contain same IP.
+    Given Minishift has state "Running"
+     When executing "minishift console --url"
+     Then stdout should be valid address
+      And "minishift ip" should return same ip
+
+  Scenario: OpenShift is ready after startup
+    After startup of Minishift OpenShift instance should respond correctly on its html endpoints
+    and OpenShift web console should be accessible.
+    Given Minishift has state "Running"
+     Then response of OpenShift at "/healthz" contains "ok"
```

Comment:
Interesting with nested steps. Need to better understand how this works. I am not quite sure how this would work in our case. Could this nested step approach not conflict in some cases?

> Second option is to implement the replacement of ${var} only specified steps.

Right, that was my thinking initially.

> Found out this to be doable, we just need to update godog package from 0.6.2 to 0.7.1 in glide.yaml.

An update is probably good regardless. There might be other fixes we get via this.

> I had a conversation about this with l3pp4rd, contributor of godog:
>> But it looks quite misleading to make such dynamic implementations in step level. I do not consider this as a good practice.

I agree, I also don't consider this good practice. As I mentioned before it introduces some sort of magic. The problem with say the HTTP step is, that it is not only the path which matters for us. There is also the app part which comes in front of the routing suffix, http://\<app\>.\<project\>.\<routing-suffix\>/...

Or take 'minishift ip' case. How do I handle this. I guess one can create dedicated steps for all these cases, a bit what you had:

     Then "minishift ip" should match the Minishift VM IP

So for all cases where we would need some dynamic part, we would need to have a dedicated step which under the hood do the checks. You still need to keep information like IP, routing suffix, etc in some sort of map so.